lib/pbench/server/api/resources/__init__.py

```diff
@@ -631,6 +631,37 @@ def convert_access(value: str, parameter: "Parameter") -> str:
     return v
 
 
+def convert_boolean(value: Any, parameter: "Parameter") -> bool:
+    """Normalize a boolean parameter value
+
+    This can handle a true boolean True or False that might be presented in a
+    JSON body, or one of a set of string values which can appear in the query
+    string of a URI.
+
+    A boolean parameter value of "" signifies "true" by default: this will be
+    the Flask representation of a 'mine' keyword (without a value) appearing
+    in a URI, like `GET /datasets/list?mine`.
+
+    Args:
+        value : parameter value
+        parameter : The Parameter definition
+
+    Raises:
+        ConversionError : input can't be validated or normalized
+    """
+    v = value
+    if isinstance(value, str):
+        if v.lower() in ("", "t", "true", "y", "yes"):
+            v = True
+        elif v.lower() in ("f", "false", "n", "no"):
+            v = False
+        else:
+            raise ConversionError(value, "boolean")
+    elif not isinstance(value, bool):
+        raise ConversionError(value, "boolean")
+    return v
```

Comment:
I think the use of `v` vs. `value` is a little inconsistent, here.  If we're going to use `value` at line 653 (which I think is reasonable), then we should also use `value` at lines 654 and 656 when invoking `lower()`.  (That is, we initialize `v` to the value of `value` at line 652 in order to cover the case where we fall through to line 662; at line 653, having established that `value` is a `str` we should then use _`value`_ to invoke `str.lower()`, the result of which is used to set the return value,  `v`.)

Reply:
Alternately, dispense with the local variable (and common return point) and collect the failures in a common `raise`:
```
    if isinstance(value, bool):
        return value

    if isinstance(value, str):
        if value.lower() in ("", "t", "true", "y", "yes"):
            return True
        if value.lower() in ("f", "false", "n", "no"):
            return False

    raise ConversionError(value, "boolean")
```

Reply:
OK to the "consistency" argument, I suppose, though the names are just aliases for the same value until I reassign (and after that there's just the `return`). But I prefer the multiple raises to the multiple returns, which I find harder to read.

Reply:
I looked at it this way:  `value` is in _input_, `v` is an `output`...so we shouldn't really be **reading** `v`.  (But, that point of view didn't crystalize until after I posted the review....)

> I prefer the multiple raises to the multiple returns, which I find harder to read.

Fair enough.